pkg/controller/controller_test.go

```diff
@@ -40,3 +45,51 @@ func TestInitResource(t *testing.T) {
 		t.Errorf("initResource() failed to update existing thirdpartyresource")
 	}
 }
+
+func TestCollectObjects(t *testing.T) {
+	var (
+		nonExistFunc types.UID = "00000000-0000-0000-0000-000000000000"
+		bt                     = true
+	)
+
+	myNsFoo := metav1.ObjectMeta{
+		Namespace: "myns",
+		Name:      "foo",
+		OwnerReferences: []metav1.OwnerReference{
+			{
+				Kind:               "Function",
+				APIVersion:         "k8s.io",
+				Name:               "funcFoo",
+				UID:                nonExistFunc,
+				BlockOwnerDeletion: &bt,
+			},
+		},
+	}
+
+	deploy := xv1beta1.Deployment{
+		ObjectMeta: myNsFoo,
+	}
+
+	svc := v1.Service{
+		ObjectMeta: myNsFoo,
+	}
+
+	cm := v1.ConfigMap{
+		ObjectMeta: myNsFoo,
+	}
+
+	client := fake.NewSimpleClientset(&deploy, &svc, &cm)
+
+	functionUIDSet := make(map[types.UID]bool)
+	//functionUIDSet[nonExistFunc] = false
+
+	if err := collectServices(client, functionUIDSet); err != nil {
+		t.Errorf("collectService() failed with %v", err)
+	}
+	if err := collectDeployment(client, functionUIDSet); err != nil {
+		t.Errorf("collectDeployment() failed with %v", err)
+	}
+	if err := collectConfigMap(client, functionUIDSet); err != nil {
+		t.Errorf("collectConfigMap() failed with %v", err)
+	}
```

Comment:
Verify that there were appropriate "delete" actions called on the client.  (See example in other tests)

Ideally you would call this with an existing Function, and a mix of resources from that Function and from a different (deleted) Function.  Then you would check that the right delete operations were called for the missing Function, but not for the existing Function.